File: Homework/themoviedb/src/main/java/com/example/themoviedb/models/Movie.java
package com.example.themoviedb.models;

import android.widget.ImageView;

import com.google.gson.annotations.SerializedName;
import com.squareup.picasso.Picasso;

public class Movie {

    private final String imageLink = "https://image.tmdb.org/t/p/w185/";

    @SerializedName("poster_path")
    public String posterPath;

    @SerializedName("original_title")
    public String title;

    public Integer id;

    public boolean rating = false;

    public boolean isWatched = false;

    public void setPoster(ImageView imageView){ Picasso.get().load(this.imageLink + posterPath).into(imageView); }
}

File: Homework/themoviedb/src/main/java/com/example/themoviedb/persistence/Movie.kt
package com.example.themoviedb.persistence

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.themoviedb.models.MovieModel

@Entity(tableName = "movies")
data class Movie (
    @PrimaryKey
    var id: Int? = 0,
    var posterPath: String? = null,
    var title: String? = null,
    var rating: Boolean = false,
    var isWatched: Boolean = false
)
{
    constructor() : this(0, "", "", false, false)
    constructor(model: MovieModel) : this(0, model.posterPath, model.title, false, false)
}